build-deps.zsh

```diff
@@ -85,9 +85,9 @@ package() {
   if [[ ${PACKAGE_NAME} != 'qt'* ]] {
     log_status "Cleanup unnecessary files"
 
-    rm -rf lib/^(*.dylib|libajantv*|*.a|*.so*|*.lib|*.framework|*.dSYM|cmake)(N)
+    rm -rf lib/^(*.dylib|libajantv*|*.a|*.so*|*.lib|*.framework|*.dSYM|carla|cmake)(N)
     rm -rf lib/(libpcre*|libpng*)(N)
-    rm -rf bin/^(*.exe|*.dll|*.pdb|swig)(N)
+    rm -rf bin/^(*.exe|*.dll|*.pdb|carla-*|swig)(N)
```

Comment:
the plugin discovery and bridging. there is no way to do "out of process" handling without the use of external tools.
so there is one for finding plugin information, and another for loading and processing audio.

Reply:
err there is literally no other way to do it. we want plugins to be loaded in a different process so that OBS is safe against plugin related crashes, for that we need to start an external tool/process and setup communication channels with that process.

my initial approach to "audio plugins with OBS" was to build the entire thing as part of OBS, so the entire thing could be built, validated and tested in 1 single place.
when doing the hosting part as library provided by obs-deps, we then need to ship those external tools.

Reply:
As I commented in the companion PR, for macOS this would optimally be a Framework (see Sparkle), because then the library as well as its binaries are contained in a single bundle and Xcode would just copy it over, codesign everything, it'd be neat.

For Windows I'd probably be best to have it packaged like the stuff we build for `win-capture` (with the helper binaries moved to the plugin's "data" directory).

For Linux this might probably require having Carla built and installed globally (so the binaries sit in `/usr/bin` anyway).

The macOS part would need to be handled by its own CMake build system, the Windows part would need adjusting the copy step on `obs-studio` (though I'm not sure how the binaries will find the DLL then).

Reply:
So as an example the Framework would look like:

```
Carla.framework
 - Versions
   - A
     - Headers
       - <All Public Header Files>
     - Resources
       - <All Additional Resource Files, usually loaded at runtime>
     - Libraries
       - libcarla_utils.dylib
     - Carla // The actual Carla library
     - carla-bridge-native
     - carla-discovery-native
```

This expects the Carla Framework to be the main point of interaction with consumers, which itself then loads the binaries and the utility library. `Carla` itself would need to use an `rpath` value of `@executable_path/Libraries` to find its dependencies (same as the helper binaries).

Reply:
> The only thing I'm not sure of is how to get them to also find the associated DLLs. Windows does have the concept of DLL search paths, but those need to be set at compile time.

Just to give a quick comment to this, the carla binary tools are 100% self-contained and do not need the extra DLL.
So the case of just putting them in a data dir is fine.

I will hold on with going for that approach until it is confirmed wanted. Need to handle the macOS lib framework thing first anyhow

Reply:
In that case we'd usually copy the Carla DLLs into the `obs-plugins` directory so they sit right next to the plugin itself (just like with CEF) and the binaries (given that they are "helpers") would be put into the plugin's data directory on Windows.

That'd at least mimic the approach we chose for `obs-browser` (runtime dependencies) and `win-capture` (helper applications).

Reply:
IIRC Windows always tries to resolve DLLs in the current working directory. That's how code injections and code loaders work on Windows:

You just add a system DLL that you know will be loaded into the same directory and you can hijack the binary.